pydy/system.py

```diff
@@ -340,22 +340,24 @@ def times(self):
         equations of motion are integrated, numerically.
 
         The object should be in a format which the integration module to be
-        used can accept. Since this attribute is not checked for
-        compatibility, the user becomes responsible to supply it correctly.
+        used can accept.
         """
         return self._times
 
     @times.setter
     def times(self, new_times):
+        self._check_times(new_times)
         self._times = new_times
 
     def _check_times(self, times):
-        """
-        Very basic checking.
-        TODO: add more checking
-        """
         if len(times) == 0:
             raise TypeError("Times supplied should be in an array_like format.")
+
+        if not all(time >= 0 for time in times):
```

Comment:
`times` should be converted to a numpy array immediately and then these checks can avoid loops.